evalml/pipelines/utils.py

```diff
@@ -151,17 +169,99 @@ def _get_preprocessing_components(
             import_or_raise(
                 "imblearn.over_sampling", error_msg="imbalanced-learn is not installed"
             )
-            pp_components.append(sampler_components[sampler_name])
+            components.append(sampler_components[sampler_name])
         except ImportError:
             logger.warning(
                 "Could not import imblearn.over_sampling, so defaulting to use Undersampler"
             )
-            pp_components.append(Undersampler)
+            components.append(Undersampler)
+    return components
 
+
+def _get_standard_scaler(X, y, problem_type, estimator_class, sampler_name=None):
+    components = []
     if estimator_class and estimator_class.model_family == ModelFamily.LINEAR_MODEL:
-        pp_components.append(StandardScaler)
+        components.append(StandardScaler)
+    return components
+
+
+def _get_time_series_featurizer(X, y, problem_type, estimator_class, sampler_name=None):
+    components = []
+    if (
+        is_time_series(problem_type)
+        and estimator_class.model_family != ModelFamily.ARIMA
+    ):
+        components.append(DelayedFeatureTransformer)
+    return components
 
-    return pp_components
+
+def _get_preprocessing_components(
+    X, y, problem_type, estimator_class, sampler_name=None
+):
+    """Given input data, target data and an estimator class, construct a recommended preprocessing chain to be combined with the estimator and trained on the provided data.
+
+    Args:
+        X (pd.DataFrame): The input data of shape [n_samples, n_features].
+        y (pd.Series): The target data of length [n_samples].
+        problem_type (ProblemTypes or str): Problem type.
+        estimator_class (class): A class which subclasses Estimator estimator for pipeline.
+        sampler_name (str): The name of the sampler component to add to the pipeline. Defaults to None.
+
+    Returns:
+        list[Transformer]: A list of applicable preprocessing components to use with the estimator.
+    """
+    if is_time_series(problem_type):
```

Comment:
This is a refactor aimed at letting us tweak the pipeline structure for time series without making the implementation for non-time series pipelines more confusing. I suspect we'll need something like this too for #2511